core/drivers/atmel_saic.c

```diff
@@ -0,0 +1,360 @@
+		*src_type = AT91_AIC_SMR_HIGH_LEVEL;
+		break;
+	case IRQ_TYPE_LEVEL_LOW:
+		if (!is_external_it(it))
+			return DT_INFO_INVALID_INTERRUPT;
+
+		*src_type = AT91_AIC_SMR_LEVEL;
+		break;
+	default:
+		return DT_INFO_INVALID_INTERRUPT;
+	}
+
+	return 0;
+}
+
+static void saic_add(struct itr_chip *chip __unused, size_t it,
+		     uint32_t type, uint32_t prio)
+{
+	uint32_t src_type = AT91_AIC_SMR_HIGH_LEVEL;
+	uint32_t res = TEE_ERROR_GENERIC;
+
+	if (it >= saic.nr_irqs)
+		panic();
+
+	res = saic_get_src_type(type, it, &src_type);
+	if (res)
+		panic("Invalid interrupt specifier");
+
+	saic_configure_it(it, src_type, prio);
+}
+
+static void saic_enable(struct itr_chip *chip __unused, size_t it)
+{
+	saic_select_it(it);
+	saic_write_reg(AT91_AIC_IECR, 1);
+}
+
+static void saic_disable(struct itr_chip *chip __unused, size_t it)
+{
+	saic_select_it(it);
+	saic_write_reg(AT91_AIC_IDCR, 1);
+}
+
+static void saic_raise_pi(struct itr_chip *chip __unused, size_t it __unused)
+{
+	panic();
+}
+
+static void saic_raise_sgi(struct itr_chip *chip __unused, size_t it __unused,
+			   uint8_t cpu_mask __unused)
+{
+	panic();
+}
+
+static void saic_set_affinity(struct itr_chip *chip __unused,
+			      size_t it __unused, uint8_t cpu_mask __unused)
+{
+	panic();
+}
+
+static const struct itr_ops saic_ops = {
+	.add = saic_add,
+	.enable = saic_enable,
+	.disable = saic_disable,
+	.raise_pi = saic_raise_pi,
+	.raise_sgi = saic_raise_sgi,
+	.set_affinity = saic_set_affinity,
+};
+
+static int saic_dt_get_irq(const uint32_t *properties, int len,
+			   uint32_t *type, uint32_t *prio)
+{
+	uint32_t res = DT_INFO_INVALID_INTERRUPT;
+	int it = DT_INFO_INVALID_INTERRUPT;
+	uint32_t src_type = 0;
+	int priority = 0;
+	uint32_t irq_type = 0;
+
+	len /= sizeof(uint32_t);
+
+	if (len != 3)
+		return DT_INFO_INVALID_INTERRUPT;
+
+	it = fdt32_to_cpu(properties[0]);
+	if (it >= (int)saic.nr_irqs)
+		return DT_INFO_INVALID_INTERRUPT;
+
+	irq_type = fdt32_to_cpu(properties[1]);
+	res = saic_get_src_type(irq_type, it, &src_type);
+	if (res)
+		return res;
+
+	priority = fdt32_to_cpu(properties[2]);
+	if (priority >= AT91_AIC_MAX_PRIO)
+		return DT_INFO_INVALID_INTERRUPT;
+
+	if (type)
+		*type = irq_type;
+
+	if (prio)
+		*prio = priority;
+
+	return it;
+}
+
+struct itr_chip saic_chip = {
+	.ops = &saic_ops,
+	.dt_get_irq = &saic_dt_get_irq,
+};
+
+static void saic_clear_aicredir(void)
+{
+	vaddr_t sfr_base = sam_sfr_base();
+	uint32_t aicredir_val = 0;
+
+	aicredir_val = io_read32(sfr_base + AT91_SFR_SN1);
+	aicredir_val ^= AT91_SFR_AICREDIR_XOR_KEY;
+	aicredir_val &= AT91_SFR_AICREDIR_KEY_MASK;
+
+	/*
+	 * We explicitly don't want to redirect secure interrupts to non secure
+	 * AIC. By default, AT91Bootstrap does so on some platforms.
+	 */
+	io_write32(sfr_base + AT91_SFR_AICREDIR, aicredir_val);
+}
+
+static void saic_init_external(const void *fdt, int node)
+{
+	int i = 0;
+	int len = 0;
+	int it_grp = 0;
+	int it_off = 0;
+	size_t it = 0xFFFFFFFF;
```

Comment:
This looks like a magic number.

Reply:
Acked and there is no need for such value.